crates/typst/src/layout/inline/mod.rs

```diff
@@ -1446,3 +1489,56 @@ fn overhang(c: char) -> f64 {
         _ => 0.0,
     }
 }
+
+/// Whether the hyphen should repeat at the start of the next line.
+fn should_repeat_hyphen(pred_line: &Line) -> bool {
+    // If the predecessor line does not end with a Dash::HardHyphen, we shall not place a hyphen at
+    // the start of the next line.
+    if pred_line.dash != Some(Dash::HardHyphen) {
+        return false;
+    }
+
+    // If there's a trimmed out space, we needn't repeat the hyphen. That's the case of a text like
+    // "... kebab é a -melhor- comida que existe", where the hyphens are a kind of emphasis marker.
+    if pred_line.trimmed.end != pred_line.end {
+        return false;
+    }
+
+    // The hyphen should repeat only in the languages that requires that feature.
+    // For more information see the discussion at https://github.com/typst/typst/issues/3235
+    if let Some(Item::Text(shape)) = pred_line.last.as_ref() {
+        match shape.lang {
+            // Lower Sorbian: see https://dolnoserbski.de/ortografija/psawidla/K3
+            //
+            // Czech: see https://prirucka.ujc.cas.cz/?id=164
+            //
+            // Croatian: see http://pravopis.hr/pravilo/spojnica/68/
+            //
+            // Polish: see https://www.ortograf.pl/zasady-pisowni/lacznik-zasady-pisowni
+            //
+            // Portuguese: see Base XX of "Acordo Ortográfico da Língua Portuguesa de 1990"
+            // https://www2.senado.leg.br/bdsf/bitstream/handle/id/508145/000997415.pdf
+            //
+            // Slovak: see https://www.zones.sk/studentske-prace/gramatika/10620-pravopis-rozdelovanie-slov/
```

Comment:
```suggestion
            // - Lower Sorbian: see https://dolnoserbski.de/ortografija/psawidla/K3
            // - Czech: see https://prirucka.ujc.cas.cz/?id=164
            // - Croatian: see http://pravopis.hr/pravilo/spojnica/68/
            // - Polish: see https://www.ortograf.pl/zasady-pisowni/lacznik-zasady-pisowni
            // - Portuguese: see https://www2.senado.leg.br/bdsf/bitstream/handle/id/508145/000997415.pdf (Base XX)
            // - Slovak: see https://www.zones.sk/studentske-prace/gramatika/10620-pravopis-rozdelovanie-slov/
```
(the URLs can of course exceed 80 columns)

Reply:
Done!